### ISSUE TITLE
[0.10-dev]Crash

### ISSUE THREAD

**Issue by forchain** [Qitmeer/qitmeer#496]
```
2021-06-07|13:29:16.713 [ERROR] Failed to process block             module=SYNC       hash=00000052c7bd0c8b3ae19dd285d410d5a2e279af60a449279ab8c2a2ab7b4b5b error="Irreparable error![00000052c7bd0c8b3ae19dd285d410d5a2e279af60a449279ab8c2a2ab7b4b5b]"
2021-06-07|13:29:16.915 [INFO ] 16Uiu2HAkxH285b3xo2wxp3A1rodskqj8kwJUWyTmYdbLjT8uP8FH direction:Outbound multiAddr:/ip4/103.29.70.78/tcp/28130/p2p/16Uiu2HAkxH285b3xo2wxp3A1rodskqj8kwJUWyTmYdbLjT8uP8FH activePeers:2 Peer Connected module=SYNC
panic: DAG can't find intersection!

goroutine 113 [running]:
github.com/Qitmeer/qitmeer/core/blockdag.(*Phantom).updateMainChain(0xc0006da1e0, 0xc002767940, 0xc002767940, 0xc002767940)
        /home/fiona/github.com/qitmeer/0.10-dev/core/blockdag/phantom.go:286 +0x3eb
github.com/Qitmeer/qitmeer/core/blockdag.(*Phantom).AddBlock(0xc0006da1e0, 0x15e4740, 0xc002767940, 0x0)
        /home/fiona/github.com/qitmeer/0.10-dev/core/blockdag/phantom.go:65 +0xb3
github.com/Qitmeer/qitmeer/core/blockdag.(*BlockDAG).AddBlock(0xc000730000, 0x15cfd40, 0xc00239e5a0, 0x0, 0x0, 0x0, 0x0)
        /home/fiona/github.com/qitmeer/0.10-dev/core/blockdag/blockdag.go:322 +0x589
github.com/Qitmeer/qitmeer/core/blockchain.(*BlockChain).maybeAcceptBlock(0xc00072c000, 0xc0020cf8f0, 0x4, 0x0, 0x0)
        /home/fiona/github.com/qitmeer/0.10-dev/core/blockchain/accept.go:129 +0x46d
github.com/Qitmeer/qitmeer/core/blockchain.(*BlockChain).ProcessBlock(0xc00072c000, 0xc0020cf8f0, 0x4, 0xc0020cf8f0, 0x0, 0x0)
        /home/fiona/github.com/qitmeer/0.10-dev/core/blockchain/process.go:139 +0x4a8
github.com/Qitmeer/qitmeer/p2p/synch.(*PeerSync).processGetBlockDatas(0xc000260100, 0xc0007301e0, 0xc00028ec00, 0x69, 0x80, 0x15b3040, 0xc00007ba10)
        /home/fiona/github.com/qitmeer/0.10-dev/p2p/synch/getblockdatas.go:226 +0x507
github.com/Qitmeer/qitmeer/p2p/synch.(*PeerSync).handler(0xc000260100)
        /home/fiona/github.com/qitmeer/0.10-dev/p2p/synch/peersync.go:96 +0xa74
created by github.com/Qitmeer/qitmeer/p2p/synch.(*PeerSync).Start
        /home/fiona/github.com/qitmeer/0.10-dev/p2p/synch/peersync.go:55 +0x10f

```

**Comment by forchain** [comment]
data

https://cowtransfer.com/s/b90f705c8d044c
608178

**Comment by forchain** [comment]
logs

https://cowtransfer.com/s/c743df979f1c4c
951137

**Comment by dindinw** [comment]
fixed by #498
